lib/client.js

```diff
@@ -9,6 +9,18 @@ const SLEEP_FOR = 1000;
 const MAX_RETRIES = 3;
 const MAX_PER_MINUTE = 0;
 
+// LibUV error codes we should attempt to resolve by retrying.
+// List of all codes: https://github.com/nodejs/node/blob/8174d0c8cae857296c45b2c448348f2c781f6ace/deps/uv/include/uv.h#L66-L145
+const RETRYABLE_ERRORS = [
```

Comment:
These seem reasonable to me. The only other ones that I've seen are "network down" errors, which sometimes seem to work by taking a different path through the intertubes:
```
  'ENETDOWN',       // network is down
  'ENETUNREACH',    // network is unreachable
```

however, I think it's a far better idea to _not_ add these, instead favoring what you've done here, which is carving out the list of errors that have a high likelihood of occurring, and only writing / maintaining code to handle those. We could be writing code that handles [418 responses](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418), but that's just not enough bang-for-buck.

Reply:
STOP THE PRESSES! I can’t believe I haven’t written a special handler here for servers that are actually teapots. I think I’d better actually throw a new custom error type for that one 😜

More seriously, thanks for the feedback. Glad to have this approach validated.